internal/setup/setup_test.go

```diff
@@ -1063,6 +1063,197 @@ var setupTests = []setupTest{{
 		`,
 	},
 	relerror: `invalid slice definition filename: "a.yaml"`,
+}, {
+	summary: "Package essentials with same package slice",
+	input: map[string]string{
+		"slices/mydir/mypkg.yaml": `
+			package: mypkg
+			essential:
+				- mypkg_essential
+			slices:
+				slice1:
+					contents:
+						/file:
+						/foo/bar/baz:
+				slice2:
+					contents:
+						/d**:
+				essential:
```

Comment:
"sllice3" would be much better here, because "essential" is the actual field name that we use both above and inside slices, but here it's just a slice name. The example makes that not obvious.

It's also not clear why the example has three slices. The first and second look equivalent for this example.

Also not clear why the content of the slices is so unusual.  Maybe it's a copy and paste from an unrelated test, but we should avoid that kind of thing, as it makes the tests larger and less clear. 

Finally, looks like none of the tests are covering slices with essentials?  We definitely want that in.

Reply:
I tried to make the test contain some entries but you are right they are unrelated and only clutter the tests. I have changed them to contain only what's strictly necessary.